### PR TITLE
GCW-3426 Remove some set of fields from box and pallet

### PR DESCRIPTION
**Pull request by shreyas098**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3426

### What does this PR do?

FEATURE: Remove set of fields from box and pallet

### CODE REVIEW

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 1]
```diff
@@ -724,7 +728,8 @@ def test_packages_location_changes(package)
       }
 
       describe "creating package with detail" do
-        it "creates package with detail" do
+        it "creates package with detail if storage type is package" do
```
Comment (bharat619) [inline spec/controllers/api/v1/packages_controller_spec.rb:731]: Have it something like this
```ruby
context "when package" do
  it "creates package with detail if storage type is package" do
     ....
  end
end
```

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 2]
```diff
@@ -733,6 +738,20 @@ def test_packages_location_changes(package)
           expect(parsed_body["package"]["detail_id"]).to eq(package.detail_id)
         end
 
+        it "does not creates package with detail if storage type is Box" do
```
Comment (bharat619) [inline spec/controllers/api/v1/packages_controller_spec.rb:741]: same as mentioned above

**Review by bharat619 — APPROVED** [review]

File: app/controllers/api/v1/packages_controller.rb [hunk 3]
```diff
@@ -494,6 +494,8 @@ def barcode_service
       end
 
       def assign_detail
+        return if %w[Box Pallet].include?(@package.storage_type_name)
```
Comment (swatijadhav) [inline app/controllers/api/v1/packages_controller.rb:497]:
```suggestion
        return if @package.box_or_pallet?
```

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 4]
```diff
@@ -723,17 +727,38 @@ def test_packages_location_changes(package)
         })
       }
 
-      describe "creating package with detail" do
-        it "creates package with detail" do
+      context 'if storage type is Package' do
+        it "does not creates package with detail " do
+          package_params_with_details[:storage_type]="Package"
```
Comment (swatijadhav) [inline spec/controllers/api/v1/packages_controller_spec.rb:732]:
```suggestion
          package_params_with_details[:storage_type] = "Package"
```

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 5]
```diff
@@ -723,17 +727,38 @@ def test_packages_location_changes(package)
...
           package = Package.last
           expect(parsed_body["package"]["id"]).to eq(package.id)
           expect(parsed_body["package"]["detail_type"]).to eq(package.detail_type)
           expect(parsed_body["package"]["detail_id"]).to eq(package.detail_id)
         end
+      end
+
+      context 'if storage type is Box' do
+        it "does not creates package with detail " do
+          package_params_with_details[:storage_type]="Box"
```
Comment (swatijadhav) [inline spec/controllers/api/v1/packages_controller_spec.rb:744]:
```suggestion
          package_params_with_details[:storage_type] = "Box"
```

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 6]
```diff
@@ -723,17 +727,38 @@ def test_packages_location_changes(package)
...
+          post :create, params: { package: package_params_with_details }
+          expect(response.status).to eq(422)
+          expect(parsed_body["errors"]).to_not be_nil
+        end
+      end
+
+      context 'if storage type is Pallet' do
+        it "does not creates package with detail " do
+          package_params_with_details[:storage_type]="Pallet"
```
Comment (swatijadhav) [inline spec/controllers/api/v1/packages_controller_spec.rb:753]:
```suggestion
          package_params_with_details[:storage_type] = "Pallet"
```

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 7]
```diff
@@ -723,17 +727,38 @@ def test_packages_location_changes(package)
         })
       }
 
-      describe "creating package with detail" do
-        it "creates package with detail" do
+      context 'if storage type is Package' do
+        it "does not creates package with detail " do
```
Comment (swatijadhav) [inline spec/controllers/api/v1/packages_controller_spec.rb:731]:
```suggestion
        it "creates a package with detail " do
```

File: spec/controllers/api/v1/packages_controller_spec.rb [hunk 8]
```diff
@@ -723,17 +727,38 @@ def test_packages_location_changes(package)
...
+          package_params_with_details[:storage_type]="Pallet"
+          post :create, params: { package: package_params_with_details }
+          expect(response.status).to eq(422)
+          expect(parsed_body["errors"]).to_not be_nil
+        end
+      end
 
-        it "does not create package with detail if anything fails in package" do
+      context 'if anything fails in package' do
```
Comment (swatijadhav) [inline spec/controllers/api/v1/packages_controller_spec.rb:760]:
```suggestion
      context 'if package params are incorrect' do
```

**Review by swatijadhav — APPROVED** [review]

**Pull request merged**